pkg/network/ifaceutil/interface.go

```diff
@@ -161,24 +197,67 @@ func MakeVethPair(name, master string, mtu int) (netlink.Link, error) {
 	}()
 
 	peerLink, err := netlink.LinkByName(peer)
-	if err = netlink.LinkSetMaster(peerLink, masterLink); err != nil {
-		return nil, err
+	if err != nil {
+		return nil, fmt.Errorf("failed to get peer link %q: %v", peer, err)
 	}
 
-	// make sure the lowerhalf is up, this automatically sets the upperhalf UP
-	if err = netlink.LinkSetUp(peerLink); err != nil {
-		return nil, errors.Wrap(err, "could not set veth peer up")
+	if err := netlink.LinkSetMaster(peerLink, masterLink); err != nil {
+		return nil, fmt.Errorf("failed to set master for peer link %q: %w", peer, err)
 	}
 
 	// Re-fetch the link to get its creation-time parameters, e.g. index and mac
-	veth2, err := netlink.LinkByName(name)
+	// either from the host, or from the namespace
+	var veth2 netlink.Link
+
+	veth2, err = netlink.LinkByName(name)
```

Comment:
Does a link with name `name` even exist here? Note that so far the 2 links that exists are `ifName` and `peer`, not `name`!

Also `veth2` is not a good varible name. i don't know what you refer too.